docs/RationaleSV.md

```diff
@@ -145,3 +145,20 @@ sv.verbatim "MACRO({{0}}, {{1}} reg={{4}}, {{3}})"
 The SV dialect is primarily designed for human consumption, not machines.  As
 such, transformations should aim to reduce redundancy, eliminate useless
 constructs (e.g. eliminate empty ifdef and if blocks), etc.
+
+## Attributes
+### Namings
+- `sv.namehint` - Used by ExportVerilog. If the associated value is required to have a name, the specified name will be used. This attribute is just a hint so it is not guranteed that the specified name appears in the output verilog. Example,
+```mlir
+%0 = sv.wire {sv.namehint = "foo"} : i1
+// wire foo;
+%1 = comb.add %a, %b {sv.namehint = "bar"} : i1
+// If a temporary wire is created:
+//   wire bar = a + b;
+%dead_expr = comb.and %a, %a {sv.namehint = "baz"} : i1
+// sv.namehint doesn't prevent any optimization. %dead_expr will be erased. To prevent an optimization
+```
+- `sv.user_specified_name` - Used by PrettifyVerilog. This attribute can be attached to sv.wire. At PrettifyVerilog, if the wire is dead (aka. no users), the wire will be deleted even if the wire has a symbol.
```

Comment:
no.  This defeats the purpose of a symbol.  If you don't need the wire to appear in the output, strip the symbol.

Reply:
It makes sense. Note that it is necessary to switch the behavior of several SV canonicalizers based on the appearance of the attribute if we don't use a symbol.

Reply:
Could this be handled then in LowerToHW, when it's deciding to add a symbol for name preservation? Instead of adding this unit attr, move the check currently in Prettify into LowerToHW and gate adding a symbol based on that?

Reply:
That's a good point. Probably FIRRTL canonicalizer can delete dead wires/nodes created by name preservations. If doing so, it would not be required to change LowerToHW (because if named wires/nodes reach to LowerToHW, it means they are alive).

Reply:
If done at the canonicalizer, would that lead to this situation?

> With allowing the canonicaliztion:
>
>  assign b = a; // foo.fir:2:10

I'm not sure if that would happen, but I was thinking doing it at LowerToHW might avoid this.

There's also an argument that we should be doing that anyway...

Reply:
I think we are currently not named nodes at FIRRTL even when nodes are dead because it was expected behavior of name preservation. However, if it is allowed to eliminate dead named nodes, I guess we can add that canonicalization.

Reply:
I see, well if it can be done as a FIRRTL canonicalization, I think that makes sense.